### PR TITLE
Myprofile patch

### LINKED ISSUE #436: User Profile Change API
PATCH API endpoint for updating user Profile.

Attention: For Supervisor and OIC the API will take in a User ID but it will store a Profile ID, while serializing for retrieve.. we need to serializer a UserID again.

### PR DESCRIPTION
**Pull request by ilcsik**
Resolves #436

### CODE REVIEW

**Comment by robertavram** [comment]
@ilcsik did you mean to make this PR agains Develop?

**Comment by ilcsik** [comment]
@robertavram I pulled in the other one so that I have the tests, but you can merge against develop safely.

**Comment by ilcsik** [comment]
@robertavram changed the base to develop

**Comment by robertavram** [comment]
@ilcsik we'll be changing the Supervisor and  OIC. @tejfel figured out the problem with ForeignKey to User. @ntrncic we need to update your set OIC admin stuff as well

File: EquiTrack/users/views.py [hunk 1]
```diff
@@ -89,6 +92,47 @@ def get_queryset(self):
...
+        if serializer.data["oic"]:
+            serializer.data["oic"] = instance.oic.user.id
+        return Response(serializer.data)
+
+    def update(self, request, **kwargs):
+        """
+        Updates a UserProfile object.
+        Replaces supervisor and OIC with profile id.
+        """
+        supervisor_profile = UserProfile.objects.get(user__id=request.data["supervisor"])
```
Comment (robertavram) [inline EquiTrack/users/views.py]: @ilcsik model has changed,,, we're now using User for both supervisor and OIC

Comment (ilcsik) [inline EquiTrack/users/views.py]: Done in https://github.com/unicef/etools/pull/444/commits/ab4bb0bc33a73f0f420f2655b11bf74f34aca0e6

**Pull request merged**